src/xdist/workermanage.py

```diff
@@ -252,9 +254,9 @@ def setup(self):
             args = make_reltoroot(self.nodemanager.roots, args)
         if spec.popen:
             name = "popen-%s" % self.gateway.id
-            if hasattr(self.config, "_tmpdirhandler"):
-                basetemp = self.config._tmpdirhandler.getbasetemp()
-                option_dict["basetemp"] = str(basetemp.join(name))
+            basetemp = TempPathFactory.from_config(self.config).getbasetemp()
```

Comment:
how does this handle missing basetmp parameters - at first glance i would guess, that each process gets its own basetemp if done this way,

in addition the tempoary folder of the workers would be distinct from the directory of the managing process

for sanity this needs at least share the same root folder

Reply:
Not sure what you mean, but I've added more tests.

Reply:
The standard mode of operation as users observe does not pass the basetmp valve

As such it seems that und normal operation workers will not share a common basetmp unless you hoist the basetmp variable out of the loop

And there will still be a number of potential issues wrt creating tmpdir in the main process

Reply:
The previous `self.config._tmpdirhandler` should be the same as `TempPathFactory.from_config(self.config)` (except for that it uses the newer API), no?

The only change / improvement is that it now also works with `-p no:tmpdir` (a corner case), that previously would silently not use it / skip it.

Reply:
@blueyed you create a **new** instance each loop, and **each** instance will make a **own** numbered tmpdir at the `getbasetemp` time ...

Reply:
Not sure what "loop" you are referring to - I've added one more explicit test, which is the same with xdist master and pytest 5.3.x.